lib/systems/include/DrivebrainController.h

```diff
@@ -19,33 +19,73 @@
 //              we dont keep going from latent to not latent while driving and thus the driver gets jittered and the
 //              drivetrain will keep "powering up" and "powering down"
 
+// TODO - [x] correctly measure latency between drivebrain and MCU
+//
+//         measure latency by subtracting the latest MCU millis stamped message from the drivebrain from the 
+//         current system millis to 
+
+
+/*! \brief Drivebrain Controller for fail-safe pass-through control of the car
+           RESPONSIBILITIES AND DOCUMENTATION
+  
+    this class is the "controller" that allows for pass-through control as commanded by the drivebrain. It also calculates the 
+    latency of the most recent input and checks to see if the most recent input is still valid and has not expired. If the 
+    input has expired then it switches over to the fail-safe control mode (MODE0) to allow for safe failing even while the car
+    is driving so that we dont lose hard-braking capabilities.
+
+    This class can clear it's own fault by switching off of this operating mode and then swapping back to this operating mode. 
+    The fault clears the first time this controller gets evaluated while switch from the swapped-to mode back to this pass 
+    through mode. 
+
+    latency measurement:
+        - the latency is measured by comparing the mcu systick's millis stamp within the most recent packet received from the 
+          drivebrain to the current systick millis. if the difference is too great, we swap to MODE0 control. The drivebrain's 
+          MCU timestamp comes from the most recent packet it received from the MCU as it contains the MCU systick's millis 
+          stamp at that packet's creation.
+
+    config: 
+        - maximum allowed latency
+        - assigned controller mode (currently defaults to MODE4)
```

Comment:
is the assigned controller the regular pass-through controller, or the mode that we switch to when a latency fault occurs?

Reply:
oh nevermind, I got it